controllers/service/service_controller.go

```diff
@@ -99,16 +108,95 @@ func (r *serviceReconciler) reconcile(ctx context.Context, req ctrl.Request) err
 	if err := r.k8sClient.Get(ctx, req.NamespacedName, svc); err != nil {
 		return client.IgnoreNotFound(err)
 	}
+	if svc.Annotations[service.LoadBalancerAllocatingPortKey] == "true" {
+		r.lock.Lock()
+		if err := r.allocatedService(ctx, svc); err != nil {
```

Comment:
AllocateService has to be locked to gaurantee thread-safe since it read/writes map concurrently

Reply:
Can this comment be added to the code to ensure this doesn't get broken in the future?